### ISSUE TITLE
2nd non-default deploy command defined in devfile causes odo to report error

### ISSUE THREAD

**Issue by ajm01** [redhat-developer/odo#5452]
/kind bug

## What versions of software are you using?

**Operating System:**
```
Distributor ID: Ubuntu
Description:    Ubuntu 20.04.3 LTS
Release:        20.04
Codename:       focal
```
**Output of `odo version`:**
```
$ odo version
odo v2.5.0 (724f16e68)
```
## How did you run odo exactly?
`$ odo deploy`

with the following two deploy commands defined in the devfile:

```
  - id: deploy
    composite:
        commands:
          - build-image-stack-provided
          - outerloop-deploy
        group:
          kind: deploy
          isDefault: true
  - id: deploy-app-image
    composite:
       commands:
         - build-image-app-provided
         - outerloop-deploy
       group:
         kind: deploy
         isDefault: false
```
id: deploy is the default commands for the devfile and 
id: deploy-app-image is a second non-default command.

## Actual behavior
odo deply reports an error:

```
$ odo deploy
 ✗  more than one default deploy command found in devfile, should not happen
```

## Expected behavior
i expect to be able to call either:

`$ odo deploy`
or
`$odo deploy-app-image`

to have two slightly different build strategies occur as part of deploy.

**Comment by kadel** [comment]
This has two parts.
The first one is clearly a bug

/kind bug

If there is only one `kind: deploy` command that is set as `isDefault: true` all other `kind: deploy` commands are marked as `isDefault: false`, or don't set `isDefault` (default is `false`) than `odo deploy` should not error out.

The second part 

> `$odo deploy-app-image`
> 
> to have two slightly different build strategies occur as part of deploy.

is more a new feature. I've created a separate issue to cover this #5454

**Comment by feloy** [comment]
The bug is fixed in v3 as part of https://github.com/redhat-developer/odo/pull/5460

**Comment by kadel** [comment]
> The bug is fixed in v3 as part of #5460

This is something that we should backport into v2

**Comment by kadel** [comment]
> > The bug is fixed in v3 as part of #5460
> 
> This is something that we should backport into v2

just noticed https://github.com/redhat-developer/odo/pull/5481 :+1:  thank you @feloy

**Comment by rm3l** [comment]
Closing this issue as it has been fixed in v3 (via #5460) and backported to v2 (via #5481).

/close

The other part of the issue (#5454) will be addressed after v3 GA.
